framework/include/actions/ActionWarehouse.h

```diff
@@ -122,20 +122,23 @@ class ActionWarehouse :
     }
 
   /**
-   * Retrieve all actions in a specific type.
+   * Retrieve all actions in a specific type ordered by their names.
    */
   template <class T>
-  std::set<const T *> getActions()
+  std::vector<const T *> getActions()
     {
-      std::set<const T*> actions;
-      for (unsigned int i=0; i<_all_ptrs.size(); i++)
+      std::map<std::string, const T*> actions;
+      for (unsigned int i = 0; i < _all_ptrs.size(); ++i)
```

Comment:
I know I'm nit-picking but `unsigned int` it turns out is usually only 32-bit while the STL can store items ranging into the 64-bit number range. Does this matter? Maybe, maybe not, it's unclear based on what we learned this summer in the optimizer. Matching types is a good idea as it just makes your code more robust.